### PR TITLE
Fix redirectUrl issue when loginRedirect is empty

### LINKED ISSUE #9819: [2.x] Get a blank page after login if loginRedirect is empty and Auth.redirect is the login page.
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 2.9.3

### What you did

1. Setup AuthComponent without the `loginRedirect` option:
```php
public $components = array(
    'Auth' => array(
        'authenticate' => array(
            'Form' => array(
                'fields' => array(
                    'username' => 'email',
                    'password' => 'password',
                ),
                'passwordHasher' => 'Blowfish',
            ),
        ),
    ),
);
```

2. Access a page that contains a link to /users/login.
3. Click the link to the login page. Then `Auth.redirect` will be set to /users/login.
4. Enter email and password correctly, and submit the login n page.

### What happened

Logged in successfully. But I got a blank page.

### What you expected to happen

Logged in successfully. And I get the front page instead.

### Reason

In the above steps, the PHP excecutes the following code:
https://github.com/cakephp/cakephp/blob/f22129b9c73734b3c62dc5570d290f02d67c9bf1/lib/Cake/Controller/Component/AuthComponent.php#L743-L745

But we need to take NULL into account, as same as the following:
https://github.com/cakephp/cakephp/blob/f22129b9c73734b3c62dc5570d290f02d67c9bf1/lib/Cake/Controller/Component/AuthComponent.php#L746-L750

<del>3.x contains similar issue.</del>

### PR DESCRIPTION
**Pull request by chinpei215**
Fixes #9819

### CODE REVIEW

File: lib/Cake/Controller/Component/AuthComponent.php [hunk 1]
```diff
@@ -742,6 +742,9 @@ public function redirectUrl($url = null) {
 
 			if (Router::normalize($redir) === Router::normalize($this->loginAction)) {
 				$redir = $this->loginRedirect;
+				if (!$redir) {
```
Comment (ADmad) [inline lib/Cake/Controller/Component/AuthComponent.php]: How about using ternary `$redir = $this->loginRedirect ?$this->loginRedirect : '/'`

**Comment by chinpei215** [comment]
May I reopen this PR? :cry:

**Comment by ADmad** [comment]
@chinpei215 Woops, definitely didn't want to do that :P

**Comment by markstory** [comment]
Nice work!

**Comment by chinpei215** [comment]
Thank you.

**Pull request merged**